sklearn/metrics/_regression.py

```diff
@@ -1792,16 +1797,14 @@ def d2_pinball_score(
     )
 
     if sample_weight is None:
-        y_quantile = np.tile(
-            np.percentile(y_true, q=alpha * 100, axis=0), (len(y_true), 1)
-        )
-    else:
-        y_quantile = np.tile(
-            _weighted_percentile(
-                y_true, sample_weight=sample_weight, percentile_rank=alpha * 100
-            ),
-            (len(y_true), 1),
-        )
+        sample_weight = xp.ones([y_true.shape[0]], dtype=y_pred.dtype, device=device_)
+
+    y_quantile = xp.tile(
+        _weighted_percentile(
+            y_true, sample_weight=sample_weight, percentile_rank=alpha * 100, xp=xp
```

Comment:
Let's sync this branch with `main` and pass `average=True` to this call to ensure symmetric quantile computation. This might fix discrepancies between the weighted and unweighted cases.